hack/tools/prowjob-gen/config.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// ProwIgnoredConfig is the top-level configuration struct. Because we want to
+// store the configuration in test-infra as yaml file, we have to prevent prow
+// from trying to parse our configuration as prow configuration. Prow provides
+// the well-known `prow_ignored` key which is not parsed further by Prow.
+type ProwIgnoredConfig struct {
+	ProwIgnored Config `json:"prow_ignored"`
+}
+
+// Config is the configuration file struct.
+type Config struct {
+	Branches       map[string]BranchConfig `json:"branches"`
+	Templates      []Template              `json:"templates"`
+	VersionsMapper VersionsMapper          `json:"versions"`
+}
+
+// BranchConfig is the branch-based configuration struct.
+type BranchConfig struct {
+	Interval                            string     `json:"interval"`
+	KubekinsImage                       string     `json:"kubekinsImage"`
```

Comment:
What if we just call this Image? Are we losing anything? (I would prefer to avoid hard-coding kubekins if possible)

Reply:
ProwImage?

Reply:
I think this is all Prow :). I think I would go with either Image or TestImage or maybe ContainerImage (?)

Reply:
+1 for TestImage

Reply:
(additional context for TestImage. The container of the ProwJob Pod in which the tests are actually run is named "test")